### ISSUE TITLE
Regression issue: Infinispan and Kafka extensions stopped working

### ISSUE THREAD

**Issue by Sgitario** [quarkusio/quarkus#16386]
## Describe the bug
Using the Infinispan and Kafka extensions:

```
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-smallrye-reactive-messaging-kafka</artifactId>
</dependency>
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-infinispan-client</artifactId>
</dependency>
```

And having functionality to use Kafka consumer/producers and remote caches, stopped working using 999-SNAPSHOT. The test fails with this exception:

```
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: javax.enterprise.inject.spi.DeploymentException: org.infinispan.client.hotrod.exceptions.HotRodClientException:Request for messageId=11 returned server error (status=0x84): org.infinispan.server.hotrod.CacheNotFoundException: Default cache requested but not configured
Caused by: org.infinispan.client.hotrod.exceptions.HotRodClientException: org.infinispan.server.hotrod.CacheNotFoundException: Default cache requested but not configured
```

Without using Kafka, everything works fine.

Maybe this is related to the recent Vert.4 upgrade.

## To Reproduce
1. git clone https://github.com/Sgitario/quarkus-examples
2. cd quarkus-examples/reproducers/infinispan-fails-to-start
3. mvn clean install -> works
4. mvn clean install -Dquarkus.version=999-SNAPSHOT -> Does not work

### Configuration
```properties
# Infinispan
quarkus.infinispan-client.trust-store=src/main/resources/server.jks
quarkus.infinispan-client.trust-store-password=changeit
quarkus.infinispan-client.trust-store-type=jks

quarkus.infinispan-client.server-list=localhost:11222
quarkus.infinispan-client.use-auth=true

# Kafka
# Configure the SmallRye Kafka connector
kafka.bootstrap.servers=localhost:9092

// ...
```

### Quarkus version or git rev
999-SNAPSHOT

**Comment by cescoffier** [comment]
I don't believe anything changed around this. 
For the error message, it seems like Infinispan is not completely initialized. 

We would need someone from the infinispan team to look into it.

**Comment by karesti** [comment]
@cescoffier I get the point with @wburns

**Comment by karesti** [comment]
@cescoffier I did the PR to upgrade to latest Infinispan btw

**Comment by wburns** [comment]
So, debugging the tests the old version never even initialized Infinispan (breakpoint is never hit in the client), where as the latest build Quarkus is now initializing Infinispan. I am not familiar with the code that is referencing our cache, but it is requesting a cache with no name and isn't configured in the server.

Here is the full stack trace of the new version that is creating an Infinispan RemoteCacheManager.
```
"main@1" prio=5 tid=0x1 nid=NA runnable
  java.lang.Thread.State: RUNNABLE
	  at org.infinispan.client.hotrod.RemoteCacheManager.actualStart(RemoteCacheManager.java:341)
	  at org.infinispan.client.hotrod.RemoteCacheManager.start(RemoteCacheManager.java:335)
	  at org.infinispan.client.hotrod.RemoteCacheManager.<init>(RemoteCacheManager.java:193)
	  at org.infinispan.client.hotrod.RemoteCacheManager.<init>(RemoteCacheManager.java:150)
	  at io.quarkus.infinispan.client.runtime.InfinispanClientProducer.initialize(InfinispanClientProducer.java:66)
	  at io.quarkus.infinispan.client.runtime.InfinispanClientProducer.remoteCacheManager(InfinispanClientProducer.java:304)
	  - locked <0x25de> (a io.quarkus.infinispan.client.runtime.InfinispanClientProducer)
	  at io.quarkus.infinispan.client.runtime.InfinispanClientProducer_ProducerMethod_remoteCacheManager_b4943dc6282b3982ac4ff16d7d4370041f920ead_Bean.create(InfinispanClientProducer_ProducerMethod_remoteCacheManager_b4943dc6282b3982ac4ff16d7d4370041f920ead_Bean.zig:202)
	  at io.quarkus.infinispan.client.runtime.InfinispanClientProducer_ProducerMethod_remoteCacheManager_b4943dc6282b3982ac4ff16d7d4370041f920ead_Bean.get(InfinispanClientProducer_ProducerMethod_remoteCacheManager_b4943dc6282b3982ac4ff16d7d4370041f920ead_Bean.zig:232)
	  at io.quarkus.infinispan.client.runtime.InfinispanClientProducer_ProducerMethod_remoteCacheManager_b4943dc6282b3982ac4ff16d7d4370041f920ead_Bean.get(InfinispanClientProducer_ProducerMethod_remoteCacheManager_b4943dc6282b3982ac4ff16d7d4370041f920ead_Bean.zig:267)
	  at io.quarkus.infinispan.client.runtime.InfinispanClientProducer_ProducerMethod_getRemoteCache_bb623773b911aceb82d147299cbc08fda5d8c50e_Bean.create(InfinispanClientProducer_ProducerMethod_getRemoteCache_bb623773b911aceb82d147299cbc08fda5d8c50e_Bean.zig:617)
	  at io.quarkus.infinispan.client.runtime.InfinispanClientProducer_ProducerMethod_getRemoteCache_bb623773b911aceb82d147299cbc08fda5d8c50e_Bean.get(InfinispanClientProducer_ProducerMethod_getRemoteCache_bb623773b911aceb82d147299cbc08fda5d8c50e_Bean.zig:656)
	  at io.quarkus.infinispan.client.runtime.InfinispanClientProducer_ProducerMethod_getRemoteCache_bb623773b911aceb82d147299cbc08fda5d8c50e_Bean.get(InfinispanClientProducer_ProducerMethod_getRemoteCache_bb623773b911aceb82d147299cbc08fda5d8c50e_Bean.zig:691)
	  at io.quarkus.arc.impl.InstanceImpl.getBeanInstance(InstanceImpl.java:210)
	  at io.quarkus.arc.impl.InstanceImpl.getInternal(InstanceImpl.java:196)
	  at io.quarkus.arc.impl.InstanceImpl.get(InstanceImpl.java:93)
	  at io.smallrye.reactive.messaging.kafka.KafkaConnector.getSubscriberBuilder(KafkaConnector.java:222)
	  at io.smallrye.reactive.messaging.kafka.KafkaConnector_ClientProxy.getSubscriberBuilder(KafkaConnector_ClientProxy.zig:176)
	  at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.createSubscriberBuilder(ConfiguredChannelFactory.java:207)
	  at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.register(ConfiguredChannelFactory.java:164)
	  at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.initialize(ConfiguredChannelFactory.java:125)
	  at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory_ClientProxy.initialize(ConfiguredChannelFactory_ClientProxy.zig:237)
	  at io.smallrye.reactive.messaging.extension.MediatorManager$$Lambda$907.180425543.accept(Unknown Source:-1)
	  at java.util.Iterator.forEachRemaining(Iterator.java:133)
	  at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	  at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	  at io.smallrye.reactive.messaging.extension.MediatorManager.start(MediatorManager.java:184)
	  at io.smallrye.reactive.messaging.extension.MediatorManager_ClientProxy.start(MediatorManager_ClientProxy.zig:330)
	  at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle.onApplicationStart(SmallRyeReactiveMessagingLifecycle.java:40)
	  at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle_Observer_onApplicationStart_4e8937813d9e8faff65c3c07f88fa96615b70e70.notify(SmallRyeReactiveMessagingLifecycle_Observer_onApplicationStart_4e8937813d9e8faff65c3c07f88fa96615b70e70.zig:129)
	  at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:283)
	  at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:268)
	  at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:70)
	  at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:128)
	  at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:97)
	  at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(LifecycleEventsBuildStep$startupEvent1144526294.zig:87)
	  at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(LifecycleEventsBuildStep$startupEvent1144526294.zig:40)
	  at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:652)
	  at io.quarkus.runtime.Application.start(Application.java:90)
	  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-1)
	  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	  at java.lang.reflect.Method.invoke(Method.java:566)
	  at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:211)
	  at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:331)
	  at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:642)
	  at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:688)
	  at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$8(ClassBasedTestDescriptor.java:368)
	  at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor$$Lambda$297.1624972302.execute(Unknown Source:-1)
	  at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	  at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:368)
	  at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:192)
	  at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:136)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$255.1551446957.execute(Unknown Source:-1)
	  at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$254.2030937207.invoke(Unknown Source:-1)
	  at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$253.1710814638.execute(Unknown Source:-1)
	  at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	  at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$259.361268035.accept(Unknown Source:-1)
	  at java.util.ArrayList.forEach(ArrayList.java:1541)
	  at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$255.1551446957.execute(Unknown Source:-1)
	  at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$254.2030937207.invoke(Unknown Source:-1)
	  at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$253.1710814638.execute(Unknown Source:-1)
	  at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	  at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	  at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	  at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	  at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	  at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	  at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	  at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	  at org.junit.platform.launcher.core.EngineExecutionOrchestrator$$Lambda$217.1191654595.accept(Unknown Source:-1)
	  at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	  at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	  at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	  at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	  at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
	  at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	  at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
	  at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	  at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	  at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	  at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
```

**Comment by wburns** [comment]
My guess is there is a configuration missing for the cache name of the kafka side possibly. But as far as I can see there is no issue on the Infinispan side as this is expected behavior.

**Comment by cescoffier** [comment]
Thanks @wburns. Kafka should not touch infinispan (at least, not as far as I know). @Ladicek can it be because of the new @Identified injected? We may inject too many instance which get 'instantiated' (just a guess)

**Comment by Ladicek** [comment]
That doesn't seem related. As the error message clearly says, and @wburns confirms, the Infinispan server doesn't have the requested cache configured. Inconvenitenly, neither the issue description nor the reproducer contain any information about how to configure the Infinispan server. @Sgitario care to add more info about that?

**Comment by Sgitario** [comment]
> That doesn't seem related. As the error message clearly says, and @wburns confirms, the Infinispan server doesn't have the requested cache configured. Inconvenitenly, neither the issue description nor the reproducer contain any information about how to configure the Infinispan server. @Sgitario care to add more info about that?

It's in the reproducer... https://github.com/Sgitario/quarkus-examples/blob/main/reproducers/infinispan-fails-to-start/src/test/java/io/quarkus/qe/containers/InfinispanTestResource.java

**Comment by Ladicek** [comment]
Ayea, didn't see that. Will check.

**Comment by Ladicek** [comment]
So I found that the test programmatically creates all the caches it needs, which means the problem must be elsewhere. Then I noticed the exception is thrown during the Kafka connector initialization, which lead me to finding out that...

@cescoffier is of course correct :-)

The problem is indeed caused by my PR to move from `@Named` to `@Identifier`. Fix is simple: https://github.com/smallrye/smallrye-reactive-messaging/pull/1168

If you look at my original PR https://github.com/smallrye/smallrye-reactive-messaging/pull/1108/files#diff-299c943e7c2bca731d4a2c3700bf186a119dc0d7a9d0f31c588a4d263f1bbd82, you will note that I did it correctly in the `getPublisherBuilder` method, but messed up in the `getSubscriberBuilder` method.

Due to that mess up, we now check if there's a bean of type `Map<String, Object>` with `@Named("default-kafka-broker")` or `@Identifier("default-kafka-broker")`, and if so, we look up _any_ bean of that type (no matter the qualifiers).

Which is where Infinispan comes in, because it provides a CDI bean (actually a producer method) of type `RemoteCache<K, V>`. And since `RemoteCache <: BasicCache` <: ConcurrentMap` <: Map`, this bean also automatically gets a type of `Map<K, V>`. Aaand, boom!
